internal/generator/vector/output/config_strategy.go

```diff
@@ -46,7 +46,7 @@ func (o Output) VisitRequest(r common.Request) common.Request {
 // VisitBuffer modifies the buffer behavior depending upon the value
 // of the tuning.Delivery mode
 func (o Output) VisitBuffer(b common.Buffer) common.Buffer {
-	switch o.tuning.Delivery {
+	switch o.tuning.DeliveryMode {
```

Comment:
The same change also needs to be made in `test/helpers/outputs/adapter/fake/config_strategy.go`.